### ISSUE TITLE
R/O access user experience

### ISSUE THREAD

**Issue by VishalNehra** [TeamAmaze/AmazeFileManager#2085]
**Describe the bug**
Currently the user experience is not good in case SMB client has R/O access. User can't see it anywhere on screen, and as he performs any R/W operation, there's no 'Failed' or 'Not permitted' callback shown to the user. It just silently closes with following log.

**Expected behavior**
Can extend to SFTP client as well. Better maintain connection permission state and show somewhere in UI all the time so user doesn't perform R/W operations.

**Additional context**

```
2020-11-13 12:55:17.852 29613-31247/com.amaze.filemanager.debug W/System.err: jcifs.smb.SmbAuthException: Access is denied.
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTransportImpl.checkStatus(SmbTransportImpl.java:1376)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTransportImpl.checkStatus(SmbTransportImpl.java:1573)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTransportImpl.sendrecv(SmbTransportImpl.java:1023)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTransportImpl.send(SmbTransportImpl.java:1539)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbSessionImpl.send(SmbSessionImpl.java:409)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTreeImpl.send(SmbTreeImpl.java:472)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTreeConnection.send0(SmbTreeConnection.java:404)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTreeConnection.send(SmbTreeConnection.java:318)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTreeConnection.send(SmbTreeConnection.java:298)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbTreeHandleImpl.send(SmbTreeHandleImpl.java:130)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbFile.renameTo(SmbFile.java:1353)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at jcifs.smb.SmbFile.renameTo(SmbFile.java:1303)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at com.amaze.filemanager.filesystem.Operations$3.doInBackground(Operations.java:396)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at com.amaze.filemanager.filesystem.Operations$3.doInBackground(Operations.java:369)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at android.os.AsyncTask$3.call(AsyncTask.java:378)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
2020-11-13 12:55:17.853 29613-31247/com.amaze.filemanager.debug W/System.err:     at java.lang.Thread.run(Thread.java:919)
```

**Comment by EmmanuelMess** [comment]
This is urgent now, as API 30 is needed to push updates and also adds a lot of uneditable directories.

**Comment by TranceLove** [comment]
> This is urgent now, as API 30 is needed to push updates and also adds a lot of uneditable directories.

Not sure local access would be related to this. On Android 11 access to /Android/data is forbidden, but there is no exception thrown either.

**Comment by VishalNehra** [comment]
@TranceLove Yes. I'd like the clarification as well. @EmmanuelMess told it's a feature in Android 11 to restrict access to R/O. But how?

**Comment by TranceLove** [comment]
> @TranceLove Yes. I'd like the clarification as well. @EmmanuelMess told it's a feature in Android 11 to restrict access to R/O. But how?

Check #2015, especially [my comments left there](https://github.com/TeamAmaze/AmazeFileManager/issues/2015#issuecomment-712900067). I think there is no way to work around this unless Amaze got `MANAGE_EXTERNAL_STORAGE` usage permission from Google Play.

**Comment by TranceLove** [comment]
Closed by #2093.
